salt/modules/freebsd_update.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+'''
+Support for freebsd-update utility on FreeBSD.
+
+:maintainer:    George Mamalakis <mamalos@gmail.com>
+:maturity:      new
+:platform:      FreeBSD
+'''
+
+from __future__ import absolute_import
+
+
+# Import python libs
+import logging
+
+# Import salt libs
+import salt
+import salt.utils.decorators as decorators
+from salt.exceptions import CommandNotFoundError
+
+
+log = logging.getLogger(__name__)
+
+# Define the module's virtual name
+__virtualname__ = 'freebsd-update'
+
+
+def __virtual__():
+    '''
+    .. versionadded:: 2016.3.4
+
+    Only work on FreeBSD RELEASEs >= 6.2, where freebsd-update was introduced.
+    '''
+    if __grains__['os'] != 'FreeBSD':
+        return (False, 'The freebsd_update execution module cannot be loaded: only available on FreeBSD systems.')
+    if float(__grains__['osrelease']) < 6.2:
+        return (False, 'freebsd_update is only available on FreeBSD versions >= 6.2-RELESE')
+    if 'release' not in __grains__['kernelrelease'].lower():
+        return (False, 'freebsd_update is only available on FreeBSD RELEASES')
+    return __virtualname__
+
+
+def _cmd(**kwargs):
+    '''
+    .. versionadded:: 2016.3.4
+
+    '''
+    update_cmd = salt.utils.which('freebsd-update')
+    if not update_cmd:
+        raise CommandNotFoundError('"freebsd-update" command not found')
+
+    params = []
+    if 'basedir' in kwargs:
+        params.append('-b {0}'.format(kwargs['basedir']))
+    if 'workdir' in kwargs:
+        params.append('-d {0}'.format(kwargs['workdir']))
+    if 'conffile' in kwargs:
+        params.append('-f {0}'.format(kwargs['conffile']))
+    if 'force' in kwargs:
+        params.append('-F')
+    if 'key' in kwargs:
+        params.append('-k {0}'.format(kwargs['key']))
+    if 'newrelease' in kwargs:
+        params.append('-r {0}'.format(kwargs['newrelease']))
+    if 'server' in kwargs:
+        params.append('-s {0}'.format(kwargs['server']))
+    if 'address' in kwargs:
+        params.append('-t {0}'.format(kwargs['address']))
+
+    return '{0} {1}'.format(update_cmd, ' '.join(params))
+
+
+def _wrapper(orig, pre='', post='', err_=None, run_args={}, **kwargs):
```

Comment:
We can't have mutable types as a default. (Explanation: http://reinout.vanrees.org/weblog/2012/04/18/default-parameters.html)